### PR TITLE
ddtrace/tracer: fixed inconsistency between env and in-code options for single span matching

### PR DESCRIPTION
**Pull request by dianashevchenko**
Previously:
- using combination of env and in-code options for trace/span sampling rules would result in loss of some rules. For example, in case of using in-code options for span sampling and env variable for trace sampling, trace sampling would prevail and replace all the rules. 
- all the invalid regular expressions are escaped. In that case any invalid regex is still a valid glob. So, added a separate SpanNameServiceRule option to set specifically single span sampling rule of type 2

note:
- using combination of in-code option(WithSamplingRules) and env variables (DD_[TRACE/SPAN]_SAMPLING_RULES) can mess up the order of the indices in error message

### CODE REVIEW

**Comment by dianashevchenko** [comment]
Should probably move the `samplingRulesFromEnv` to `newConfig`, so that in-code option prevails.

**Review by ajgajg1134 — CHANGES_REQUESTED** [review]

File: ddtrace/tracer/rules_sampler.go [hunk 1]
```diff
@@ -165,6 +154,21 @@ func RateRule(rate float64) SamplingRule {
 	}
 }
 
+// SpanNameServiceRule returns a SamplingRule of type SamplingRuleSpan that applies
+// the provided sampling rate to all spans matching the operation and service name glob patterns provided.
+// Operation and service fields must be valid glob patterns. MaxPerSecond is optional and defaults to infinite, allow all behaviour.
```
[the remark this answers is not in the document]
Comment (dianashevchenko) [reply ddtrace/tracer/rules_sampler.go:159]: Or maybe this : 
```golang
// SpanNameServiceRule returns a SamplingRule of type SamplingRuleSpan that applies
// the provided sampling rate to all spans matching the operation and service name glob patterns provided.
// Operation and service fields must be valid glob patterns.
func SpanNameServiceRule(name, service string, rate float64) SamplingRule {
	return SamplingRule{
		Service:   globMatch(service),
		Name:      globMatch(name),
		Rate:      rate,
		ruleType:  SamplingRuleSpan,
		exactName: name,
		limiter:   newSingleSpanRateLimiter(0),
	}
}

// SpanNameServiceMPSRule returns a SamplingRule of type SamplingRuleSpan that applies
// the provided sampling rate to all spans matching the operation and service name glob patterns 
// up to the max number of spans per second that can be sampled. 
// Operation and service fields must be valid glob patterns.
func SpanNameServiceMPSRule(name, service string, rate, limit float64) SamplingRule {
	return SamplingRule{
		Service:      globMatch(service),
		Name:         globMatch(name),
		MaxPerSecond: limit,
		Rate:         rate,
		ruleType:     SamplingRuleSpan,
		exactName:    name,
		limiter:      newSingleSpanRateLimiter(limit),
	}
}
```

[the remark this answers is not in the document]
Comment (ajgajg1134) [reply ddtrace/tracer/rules_sampler.go:159]: That works too!

**Review by ajgajg1134 — APPROVED** [review]
Looks good! 🥳

**Pull request merged**